recipes/recastnavigation/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+
+
+class RecastNavigationConan(ConanFile):
+    name = "recastnavigation"
+    homepage = "https://github.com/recastnavigation/recastnavigation"
+    description = " Navigation-mesh Toolset for Games"
+    topics = ("conan", "navmesh", "recast", "navigation", "crowd")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "ZLIB"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob('recastnavigation-*/')[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["RECASTNAVIGATION_DEMO"] = False
+        self._cmake.definitions["RECASTNAVIGATION_TESTS"] = False
+        self._cmake.definitions["RECASTNAVIGATION_EXAMPLES"] = False
+        self._cmake.definitions["RECASTNAVIGATION_STATIC"] = True
```

Comment:
Give shared as optional.

Reply:
Recast doesn't support shared libraries because it doesn't export symbols.

Reply:
But it offers the option: https://github.com/recastnavigation/recastnavigation/blob/7bfd9a1d4caccf61e0485b2b05b29966348a8b39/Recast/CMakeLists.txt#L3

have you tried with `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS`?